browser/brave_shields/cookie_expiry_browsertest.cc

```diff
@@ -114,6 +115,28 @@ class CookieExpirationTest : public InProcessBrowserTest {
   content::ContentMockCertVerifier mock_cert_verifier_;
 };
 
+// Test that a session cookie i.e. without a max-age or expiry date is not
+// affected by the cookie expiration policy. Also, is deleted on browser close.
+IN_PROC_BROWSER_TEST_F(CookieExpirationTest, CheckExpiryForSessionCookie) {
+  GURL url = https_server_->GetURL("a.com", "/simple.html");
+  ASSERT_TRUE(ui_test_utils::NavigateToURL(browser(), url));
+  JSCookieStoreWriteCookie(browser(), "null");
+  std::vector<net::CanonicalCookie> all_cookies =
+      GetAllCookiesDirect(browser());
+  EXPECT_EQ(1u, all_cookies.size());
+  for (const net::CanonicalCookie& cookie : all_cookies) {
+    EXPECT_TRUE(cookie.ExpiryDate().is_null());
+  }
+  // Fast-forward by 8 days and check that the cookie is still there.
+  base::ScopedMockTimeMessageLoopTaskRunner scoped_mock_time_task_runner;
+  scoped_mock_time_task_runner.task_runner()->FastForwardBy(base::Days(8));
```

Comment:
you also need a test that does a similar thing, but checks that a cookie actually expires (if the expiration is set) and is no longer available.

Reply:
The rest of the tests here should do that! I wanted to add a new one for session cookies which don't have expiration set.

Reply:
Other tests here validate that the lifetime of a cookie is limited to some value. It would be actually nice to have a test that checks if a cookie expires when you fast forward the time.

Can you do that? does `scoped_mock_time_task_runner.task_runner()->FastForwardBy(base::Days(8));` actually drops such cookies? I'm curious if your test will fail if you limit the cookie lifetime.

Reply:
Ah, it seems like `GetAllCookies` gets all the cookies, even expired ones from what I can tell. Great catch. I wonder if I should use `GetCookieList` instead and check the `CookieAccessResult`.

Reply:
> Ah, it seems like `GetAllCookies` gets all the cookies, even expired ones from what I can tell. Great catch. I wonder if I should use `GetCookieList` instead and check the `CookieAccessResult`.

By bet is `scoped_mock_time_task_runner` doesn't cover everything you need to cleanup cookies.